blueprints/datamodel.tex

```diff
@@ -335,7 +335,117 @@ \subsection{Delivery Services}
+then
+
+\paragraph{Deep Caching}
+
+\paragraph{Caching}
+\begin{itemize}
+	\item \code{CACHE}
+	\item \code{RAM\_ONLY}
+	\item \code{NO\_CACHE}
+\end{itemize}
+
+%TODO - STEERING doesn't have this?
+\paragraph{Cache Servers}
+A Delivery Service's assigned Cache Servers are expressed as a set of Cache Server
+IDs.
+
+\paragraph{CDN}
+
+%TODO: use-case for check path?
+
+\paragraph{Denied Access Redirect}
+
+\paragraph{DSCP}
+
+\paragraph{ECS}
+
+\paragraph{Geographic Limiting}
+This property describes limitations to the availability of this Delivery Service's
+content on the basis of the requesting client's geographic location. It is a set of
+strings, each of which is an \href{https://www.iso.org/obp/ui/#search/code/}{ISO 3166-1}
+alpha-2 country code, optionally with ISO 3166-2 subdivisional alphabetic code. This is
+a "white list" of countries/subdivisions wherein content is to be made available\footnote{
+This property is meant to inform Traffic Router; Cache Servers cannot be relied upon to
+approve or deny access on a geographic basis. Thus, if routing is bypassed, restricted
+content is totally accessible to requesting clients.}.\\
+Content is \emph{always} available to clients whose IP addresses are found within the
+Traffic Routers' Coverage Zone File(s). With that in mind, when this property is an empty
+set it means that no geographic regions are "whitelisted" and thus \emph{only} clients
+whose IP addresses are found within a Coverage Zone File will be granted access to content.
+When this property has a "Null" type, there is no geographic restriction placed on the
+Delivery Service's content access.
+
+\paragraph{Maximum Origin Connections}
+
+%TODO: use-case for header-rewrite text?
+
+\paragraph{Miss Location}
+
+\paragraph{Name}
+A Delivery Service's "Name" is a string that uniquely identifies it among all
+Delivery Services. It MUST only contain alphanumerics, hyphens, underscores and
+spaces, and MUST NOT begin with a non-alphanumeric character nor end with a
+non-alphanumeric character. This is used to generate part of the default request
+hostnames by replacing all non-alphanumeric characters with a hyphen.
+
+\paragraph{Notes}
+This section is an arbitrary string containing miscellaneous, human-friendly
+information about the Delivery Service. Other ATC components SHOULD NOT parse
+this for specific information fields, or expect it to be in a particular format.
+
+%TODO: MSO use-case?
+\paragraph{Origin}
+
+\paragraph{Query String Handling}
+
+\begin{itemize}
+	\item \code{DROP}
+	\item \code{IGNORE}
+	\item \code{USE}
+\end{itemize}
+
+\paragraph{Range Request Handling}
+\begin{itemize}
+	\item \code{NO\_CACHE}
+	\item \code{WHOLE\_OBJECT}
+	\item \code{CACHE}
+\end{itemize}
+
+%TODO Raw-Remap use-case?
+%TODO Regex-Remap use-case?
+
+\paragraph{Required Capabilities}
+
+\paragraph{Routing Name}
+
+\paragraph{Routing Type}
+\begin{itemize}
+	\item \code{HTTP}
+	\item \code{DNS}
+	\item \code{STEERING}
+	\item \code{STATIC}
```

Comment:
By static, I assume you mean ANY_MAP.  This is not an option anyone without operator should know about because it's capable of harming traffic outside a tenant.

Reply:
Well no, actually I mean Federations. My hope is that ANY_MAP can be done away with entirely, but that's still something to be discussed.

Since the whole code base is open source, though, I see  little point in not documenting things just because a non-operator can do harm with them. In 2.2 I can count at least three ways to gain root access on a machine running TO - that wouldn't be safer if they weren't documented. That ought to just be not possible.

Reply:
ANY_MAP is unfortunately still required in dire circumstances.  But a general user shouldn't know about it.

Reply:
Okay, but this is a full data model definition, not a help guide for general user.

Like I said, we'll still need to discuss ANY_MAP because I don't actually know what "dire circumstances" necessitate it.